nilearn/glm/first_level/first_level.py

```diff
@@ -72,6 +73,28 @@ def _ar_model_fit(X, val, Y):
     return ARModel(X, val).fit(Y)
 
 
+def _yule_walker(X, order):
+    """Compute Yule-Walker (adapted from MNE and statsmodels).
+    Operates in-place.
+    """
+    from scipy import linalg
+    if order < 1:
+        raise ValueError("AR order must be positive")
+    if type(order) is not int:
+        raise TypeError("AR order must be an integer")
+
+    denom = X.shape[-1] - np.arange(order + 1)
+    r = np.zeros(order + 1, np.float64)
+    X = X.ravel()
```

Comment:
to me you should not have to ravel. It's dangerous. You should enforce the ndim is correct ie here ndim=1 on call

Reply:
That makes sense. Thanks